contexts/credentials/v2

```diff
@@ -43,17 +44,20 @@
     },
     "cnf": {
       "@id": "https://www.iana.org/assignments/jwt#cnf",
-      "@context": {
-        "@protected": true,
-        "kid": {
-          "@id": "https://www.iana.org/assignments/jwt#kid",
```

Comment:
@OR13 Just to be clear, all values you've declared as `"@type": "@id"` MUST be URIs... you cannot have non-URL values in there... are you sure you want to do this?

Reply:
JSON-LD also says that some text fields are datetime strings... my experience is that eventually you process a string that is supposed to be a URL or a datetime, and it turns out to be an emoji.... Also, google uses `@id` values that are not URLs in their enterprise knowledge graph search API, so i'm not sure I would assume that those values are URLs.

Reply:
@OR13,

What's being said here is that processors will raise errors if you use `"@type": "@id"` in the term's context definition if something other than a URL or graph node is used in the data. This is not the same case as using a datetime string that is malformed, etc., as those will not be processed by default. So, if you want to allow any kind of string in the field, you should remove `"@type": "@id"`, otherwise default processing will try to enforce the use of URLs or graph nodes by raising errors when something else is present.

Reply:
A JOSE `kid` is a string and need not be a URI.  The typing should reflect that.

Reply:
I don't think we should remove the `@type: @id` from `kid` or `iss` which can both be strings or URIs because.... this causes the RDF graph to be reliant on `issuer` and `verificationMethod` which are only defined in the context of JSON-LD.... and are always URIs.... 

Removing the ability to use `iss` and `kid` and have them treated similar to `issuer` and `verificationMethod` would undermine our ability to use the core data model as JSON-LD, this is related to the "benefits of treating the data model as JSON-LD vs JSON".

Reply:
If its possible for the `@type: @id` and String to both be legal, that would be preferred... but in absence of that, we prefer the processing of these fields with DIDs or other URLs or URNs to build useful graphs.

Reply:
> @selfissued not sure it matters since URIs are strings

URIs have a structure, they're not just arbitrary strings (which, IIUC, @selfissued is saying `kid` can be any arbitrary string).

If you make `kid` and `iss` an `type: @id`, a JSON-LD processor will throw an error when processing `kid` and `iss` if it isn't a URI... I don't think we want that behavior, it'll lead to interop issues. 

That said, if you want to enforce that `kid` and `iss` be a URI when used in VCs, then using `type: @id` is one way to do that... but again, we'd be narrowing the scope of it in that case.

Reply:
> That said, if you want to enforce that `kid` and `iss` be a URI when used in VCs, then using `type: @id` is one way to do that

As shown in the example above, the industry does not agree that `@id` must be a URI... we don't need to do anything here, here is digital bazaar's json-ld processor handling the above example without erroring:

https://v.jsld.org/FwjjEErfzWY5ogNK7HM6ZFCRsfCBtfGxfpp64361qdd2bu7zFeykxD3e4gTXmcpNrztDGpfaa4zn5BNd8Uy5znajBTBBbFpwZK8QEGKSAgaU3aF7rq3DsLmuvJdGi2FFqgmQkvPiwzULpji68V6WNcHCvKmSB136ngyXwizmqP6BY8hB3zWCKAeNpt9Y5o45rdMNyUxNsdUa9JMfoecVqB94VKAHJ2z92U5KfbaDEHdAkH56JnwjVQf1EA7r2HEDVTefgiTiz81Q1KzbSrv636XupufPGKS6CwTBTZLT5paZjcQkjiiYBMRZQWxYxc8Ji4ZzUHzLngspxERKcUheDaKvemoRGrTxktsR7bwDFm6dTkStg8baTymnbPCYiX1GbHTv7mK6aJUUeykESpwedtzGRMZTmZuhdRJ2eCyZvZTyWRe8NmNMZXeek5drfFjjpCavDuHDQiZP76iuo2xEMkuL99hgrNJbGaKC6ULpYTVtXbEC6ep9oCXqebF97jibSuEkHkZMWjEKL1hF5PXKr5dReptbxExcLeUqNVbMtASXYqrQvhR9UikDTabmX19HPnua6S3KPanmMm9Mg58VGSAqiZKw1u9S8WJ1anzBy4UpS7ANWkAnTDAAZoyxKWEQrapoSgppqzEYHDqHqg1kzXJW59cQEXn1yPYW8pwNX2XUL1Kf6EjwHLjXLR

Reply:
@OR13,

That value only works when there's a base URL (like there always is with Web pages during search crawling and all that). With VCs, there is no base URL, it's presumed to be `null`, so it would result in a relative URL being used and an error as shown here: https://tinyurl.com/pys6xa4k

So I recommend we remove `"@type": "@id"` from all of the JOSE term context definitions that can take something other than a URL.

Reply:
This concern is now tracked in https://github.com/w3c/vc-data-model/issues/1275.